mule-deployer/src/main/java/org/mule/tools/client/core/AbstractClient.java

```diff
@@ -128,4 +145,33 @@ protected void configureRequest(Invocation.Builder builder) {
 
   }
 
+  protected void configureSecurityContext(ClientBuilder builder) {
+    // Implemented in concrete classes
+  }
+
+  protected void checkResponseStatus(Response response) {
```

Comment:
I'd rather go to checkResponseStatusIsSucessful or smth like that

Reply:
this one I rather leave it for two reasons: 
1. I don't like it at all and want to actually improve this whole check logic
2. It's assumed than unless say otherwise check status should fail (unless tell otherwise) if and only if status is different from 200 family  that's why I added the other method. 
I see your point though that is better to be declarative but please let's hold until we refactor that logic